src/webots/nodes/WbBaseNode.cpp

```diff
@@ -235,28 +236,70 @@ bool WbBaseNode::isInvisibleNode() const {
   return WbWorld::instance()->viewpoint()->getInvisibleNodes().contains(const_cast<WbBaseNode *>(this));
 }
 
+QString WbBaseNode::documentationUrl() const {
+  // TODO: Hotfix, undefined method
+  return QString();
+}
+
 bool WbBaseNode::exportNodeHeader(WbVrmlWriter &writer) const {
-  if (!writer.isX3d())
+  if (writer.isX3d()) {
+    writer << "<" << x3dName() << " id=\'n" << QString::number(uniqueId()) << "\'";
+    if (isInvisibleNode())
+      writer << " render=\'false\'";
+    QStringList bookAndPage = documentationBookAndPage(WbNodeUtilities::isRobotTypeName(nodeModelName()));
```

Comment:
Sorry but I don't understand the changes in this function.
It seems that only this line (and the following) is changed in the whole function.
But then why did you also change the if block structure if it behaves exactly like before?

This is really subjective topic, but I find that previous version returning immediately in case of `!writer.isX3d()` is more readable than having to search for the else block.

Reply:
You are right, there was no reason for me to change this function. I think that I added introduced changes at some point related to URDF, reverted back but I wasn't aware I made unnecessary changes in comparison to `develop` branch.